spring-web/src/main/java/org/springframework/web/util/UrlPathHelper.java

```diff
@@ -668,6 +668,20 @@ public Map<String, String> decodePathVariables(HttpServletRequest request, Map<S
 		}
 	}
 
+	/**
+	 * Decode the given URI path variables without {@link #decodeRequestString}
+	 * it is assumed the URL path from which the variables were extracted is already decoded
+	 * through a call to {@link #getLookupPathForRequest(HttpServletRequest)}. Under certain
+	 * circumstance such as passing special character, it is the only way to keep any path
+	 * variable values passed as is.
+	 * @param vars the URI variables extracted from the URL path
+	 * @return the same Map
+	 */
+	public Map<String, String> decodePathVariablesForSpecialCharacter( Map<String, String> vars) {
+		this.setUrlDecode(false);
+		return vars;
+	}
```

Comment:
`UrlPathHelper` is a singleton and is used to determine the lookup path for all incoming requests. Changing the `urlDecode` here would affect all requests, and potentially break handling mappings.